src/float.c

```diff
@@ -41,7 +41,7 @@ string2float(
 	*value = INFINITY;
 	return 3;
     }
-    if (STRNICMP(text, "-inf", 3) == 0)
+    if (STRNICMP(text, "-inf", 4) == 0)
     {
```

Comment:
I'm not sure, but I have a feeling that this whole if thing is unnecessary.
At the very least, `str2float("-inf")` passes in the character after the `-` ("inf") and negates the value you later get.
On a related note, it's interesting that `str2float("--inf")` results in `inf`.